components/brave_wallet_ui/components/desktop/popup-modals/balance_details_modal/balance_details_modal.tsx

```diff
@@ -0,0 +1,175 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import * as React from 'react'
+import Tooltip from '@brave/leo/react/tooltip'
+
+// Queries
+import {
+  useGetBitcoinBalancesQuery //
+} from '../../../../common/slices/api.slice'
+
+// Types
+import { BraveWallet } from '../../../../constants/types'
+
+// Utils
+import { getLocale } from '../../../../../common/locale'
+import Amount from '../../../../utils/amount'
+
+// Components
+import { PopupModal } from '../index'
+import LoadingSkeleton from '../../../shared/loading-skeleton'
+
+// Styled Components
+import {
+  BalancesColumn,
+  SectionText,
+  BalanceText,
+  InfoIcon,
+  TooltipContent
+} from './balance_details_modal.style'
+import { Column, Row, VerticalDivider } from '../../../shared/style'
+
+interface Props {
+  accountId: BraveWallet.AccountId
+  token: BraveWallet.BlockchainToken
+  onClose: () => void
+}
+
+export const BalanceDetailsModal = React.forwardRef<HTMLDivElement, Props>(
+  (props: Props, forwardedRef) => {
+    const { onClose, accountId, token } = props
+
+    // Queries
+    const { data: balances, isLoading: isLoadingBalances } =
+      useGetBitcoinBalancesQuery({
+        accountID: accountId
+      })
+
```

Comment:
Maybe balance should come from props? It is already available for all uses of this component. And no need to handle loading state also.

Reply:
Good idea, moved to props!